### DISCUSSION TITLE
what is the maximum "Nodes per Second" at single core ?

### DISCUSSION THREAD

**Discussion by ghost** [official-stockfish/Stockfish#4168] [Q&A] [answered]
https://github.com/official-stockfish/Stockfish/issues/4166#issue-1373578176

_`info depth 70 seldepth 88 multipv 1 score cp 93 wdl 169 829 2 nodes 10328639573768 nps 46749714 hashfull 999`_

Based on one of the issues sent by a friend , I see **46 Million NPS** speed for **stockfish** , do this speed is really ?

what is the highest speed of NPS  at single core ? 

Thanks

**Comment by dav1312** [comment, marked as answer]
1. That is not a single core test
2. There is no hardcoded limit

**Reply by ghost** [reply]
i know there is no limit , but with the best of 2022 CPU : what is maximum speed NPS  at single core ?

if possible please share your NPS at single core in start posotion , thanks
;

**Reply by dav1312** [reply]
You should expect around 1-2Mnps

**Reply by ghost** [reply]
> You should expect around 1-2Mnps

yes I agree with Intel Core i9-12900KS at single core NPS  is about 2 Million

**Reply by RogerThiede** [reply]
https://ipmanchess.yolasite.com/amd---intel-chess-bench.php

**Reply by ghost** [reply]
> https://ipmanchess.yolasite.com/amd---intel-chess-bench.php

thanks @RogerThiede  , i see this link but test was multi thread  , may i ask you what is maximum NPS at signle thread ?

**Reply by RogerThiede** [reply]
I suspect that would depend on too many variables, most notably being that the complexity of a given FEN position can cause significant slowdowns, let alone the size of the transposition table hash and whether or not it is full due to cache misses and/or extraneous memory fetches.
Using a pure neural network evaluation more similar to Leela with no strategic search algorithm of moves would likely result in the highest nodes per second possible; however that could take longer to identify the most optimal lines even though it reaches higher NPS.  Stockfish spends additional resources on the move picker and other strategies.
You can rent various hardware architectures (BMI2, AVX2, AVX512, VNNI512, etc,) from cloud companies and see different results just based on different compilers (gcc, clang, etc.)
